treegp/two_pcf.py

```diff
@@ -0,0 +1,410 @@
+
+        :param X:  Coordinates of the field. (n_samples, 1 or 2)
+        :param y:  Values of the field. (n_samples)
+        :param y_err: Error of y. (n_samples)
+        """
+        if np.sum(y_err) == 0:
+            w = None
+        else:
+            w = 1./y_err**2
+
+        if self.anisotropic:
+            cat = treecorr.Catalog(x=X[:,0], y=X[:,1], k=(y-np.mean(y)), w=w)
+            kk = treecorr.KKCorrelation(min_sep=self.min_sep, max_sep=self.max_sep, nbins=self.nbins,
+                                        bin_type='TwoD', bin_slop=0)
+            kk.process(cat)
+            # Need a mask in the case of the 2D correlation function, to compute
+            # the covariance matrix using the bootstrap. The 2D correlation
+            # function is symmetric, so just half of the correlation function
+            # is useful to compute the covariance matrix. If it is not done,
+            # the covariance matrix is consequently not positive definite.
+            npixels = len(kk.xi)**2
+            mask = np.ones_like(kk.xi, dtype=bool)
+            mask = mask.reshape((int(np.sqrt(npixels)), int(np.sqrt(npixels))))
+
+            boolean_mask_odd = (len(mask)%2 == 0)
+            even_or_odd = len(mask)%2
+            nmask = int((len(mask)/2) + even_or_odd)
+            mask[nmask:,:] = False
+            mask[nmask-1][nmask:] = boolean_mask_odd
+
+            mask = mask.reshape(npixels)
+
+            dy = (kk.bottom_edges + kk.top_edges) / 2.
+            dx = dy.T
+
+            distance = np.array([dx.reshape(npixels), dy.reshape(npixels)]).T
+            Coord = distance
+            xi = kk.xi.reshape(npixels)
+        else:
+            cat = treecorr.Catalog(x=X[:,0], y=X[:,1], k=(y-np.mean(y)), w=w)
+            kk = treecorr.KKCorrelation(min_sep=self.min_sep, max_sep=self.max_sep, nbins=self.nbins)
+            kk.process(cat)
+            distance = kk.meanr
+            mask = np.ones_like(kk.xi, dtype=bool)
+            Coord = np.array([distance,np.zeros_like(distance)]).T
+            xi = kk.xi
+
+        return xi, distance, Coord, mask
+
+    def comp_xi_covariance(self, n_bootstrap=1000, mask=None, seed=610639139):
+        """
+        Estimate 2-point correlation function covariance matrix using Bootstrap.
+
+        :param seed: seed of the random generator.
+        """
+        np.random.seed(seed)
+        xi_bootstrap = []
+        for i in range(n_bootstrap):
+            u, v, y, y_err = self.resample_bootstrap()
+            coord = np.array([u, v]).T
+            xi, d, c, m = self.comp_2pcf(coord, y, y_err)
+            if mask is None:
+                mask = np.array([True]*len(xi))
+            xi_bootstrap.append(xi[mask])
+        xi_bootstrap = np.array(xi_bootstrap)
+
+        dxi = xi_bootstrap - np.mean(xi_bootstrap, axis=0)
+        xi_cov = 1./(len(dxi)-1.) * np.dot(dxi.T, dxi)
+        return xi_cov
+
+    def return_2pcf(self, seed=610639139):
```

Comment:
Is there a reason to set a default seed? I'd imagine that it would be preferable for the default behavior to be random and just use specific seeds for testing?

Reply:
I guess this is good to have a default fixed seed. If a user don't know that there is a random part, it might be confusing that by re-running the code, parameters are changing because of a non-fixed seed.